pkg/apiserver/archive/archive.go

```diff
@@ -278,38 +279,62 @@ func (s *Service) report(c *gin.Context) {
 // @Description Delete the specified archived experiment.
 // @Tags archives
 // @Produce json
-// @Param uid path string true "uid"
+// @Param uid path string false "uid"
+// @Param uids query string false "uids"
 // @Success 200 {object} StatusResponse
 // @Failure 500 {object} utils.APIError
 // @Router /archives/{uid} [delete]
 func (s *Service) delete(c *gin.Context) {
 	var (
-		err error
-		exp *core.Experiment
+		err      error
+		exp      *core.Experiment
+		uidSlice []string
 	)
 
 	uid := c.Param("uid")
+	uids := c.Query("uids")
+	if uids != "" {
+		uidSlice = strings.Split(uids, ",")
+	}
+	errFlag := false
 
-	if exp, err = s.archive.FindByUID(context.Background(), uid); err != nil {
-		if gorm.IsRecordNotFoundError(err) {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInvalidRequest.New("the archived experiment is not found"))
-		} else {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
-		}
+	if uid != "" && uids != "" {
+		c.Status(http.StatusBadRequest)
+		_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(fmt.Errorf("uid and uids cannot exist at the same time")))
+		return
+	} else if uid == "" && uids == "" {
+		c.Status(http.StatusBadRequest)
+		_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(fmt.Errorf("uid and uids cannot be empty at the same time")))
 		return
 	}
 
-	if err = s.archive.Delete(context.Background(), exp); err != nil {
-		c.Status(http.StatusInternalServerError)
-		_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
-	} else {
+	if uid != "" {
+		uidSlice = append(uidSlice, uid)
+	}
+	for _, uid := range uidSlice {
+		if exp, err = s.archive.FindByUID(context.Background(), uid); err != nil {
+			if gorm.IsRecordNotFoundError(err) {
+				_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(fmt.Errorf("delete archive uid (%s) error, because the archive is not found", uid)))
+			} else {
+				_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(fmt.Errorf("delete archive uid (%s) error, because %s", uid, err.Error())))
+			}
+			errFlag = true
+			continue
```

Comment:
Should replace `continue` with `break`?

Suppose there are two errors that will occur. After the range loop end, we will set the context error twice.

Reply:
The same goes for deleting experiments.

Reply:
https://pkg.go.dev/github.com/gin-gonic/gin#Context.Error